### ISSUE TITLE
"FBSSceneSnapshotAction unhandled action" warning on locking iOS 9 device

### ISSUE THREAD

**Issue by friedbunny** [mapbox/mapbox-gl-native#2172]
```objc
2015-08-24 15:28:32.262 Mapbox GL[520:125616]  ** -[UIApplication _handleNonLaunchSpecificActions:forScene:withTransitionContext:completion:] ** unhandled action -> <FBSSceneSnapshotAction: 0x14e23ac0> {
    handler = remote;
    info = <BSSettings: 0x14ed9e40> {
        (1) = 5;
    };
}
```

**Comment by micahelcheng** [comment]
I have the same issue in ios9

**Comment by aidongsheng** [comment]
i have same problem,my map app can't  work normally when the iPhone screen is locked.

**Comment by friedbunny** [comment]
@aidongsheng Can you clarify what you're expecting your map to do while the device is locked?

**Comment by RameshAran** [comment]
I have the same problem. Not sure what is the real problem.

**Comment by zmj520** [comment]
I have the problem when I lock the screen in my app.The situation is that I am trying to get the MPNowPlayingInfoCenter to set some realtime infomation,but when I lock my iPhone to see whether is successful,the Console prints the above error and the setting is invalid,here is the part of my settingCode: (environment:Xcode7.0 &iPhone6s ,development target:iOS9.0)
```objc
    // 获取锁屏中心
    MPNowPlayingInfoCenter *playingInfoCenter = [MPNowPlayingInfoCenter defaultCenter];
    // 设置锁屏信息
    NSMutableDictionary<NSString *, id> * info = [NSMutableDictionary dictionary];
    [info setObject:music.name forKey:MPMediaItemPropertyAlbumTitle]; // 歌曲名
    [info setObject:music.singer forKey:MPMediaItemPropertyArtist];// 歌手名
    [info setObject:@(self.duration) forKey:MPMediaItemPropertyPlaybackDuration]; // 时长
    MPMediaItemArtwork *artWork = [[MPMediaItemArtwork alloc] initWithImage:image];
    [info setObject:artWork forKey:MPMediaItemPropertyArtwork]; // 图片
    playingInfoCenter.nowPlayingInfo = info;
    // 让应用程序可以接受远程时间
    [[UIApplication sharedApplication] beginReceivingRemoteControlEvents];
```

**Comment by yangjw** [comment]
I have the same issue in ios9

**Comment by winkelsdorf** [comment]
Seems to be an iOS9 Bug, have a look at the Callstack ;)

    UIApplication _handleNonLaunchSpecificActions:forScene:withTransitionContext:completion:] 
    ** unhandled action -> <FBSSceneSnapshotAction:

That's an internal iOS API (UIApplication._handleNonLaunchSpecificActions) called from an internal framework, the FrontBoard Services Framework (FBS). FBSSceneSnapshotAction is obviously responsible for taking device screenshots for e.g. Multitasking-Preview on your Homescreen.

See: http://stackoverflow.com/questions/32344082/error-handlenonlaunchspecificactions-in-ios9
and: http://stackoverflow.com/questions/32658037/ios9-storyboard-what-is-unhandled-action-handlenonlaunchspecificactions

And please (!) don't implement any fix like a custom _handleNonLaunchSpecificActions for it: You will likely break Apple's internal API behavior plus you might get rejected on the App Store.

Just wait for Apple to fix it. It seems to be related to locking/unlocking in iOS9 (Bug still exists as of iOS 9.1).

**tl;dr**
NOT related to MapBox - you just found an iOS Bug

**Comment by rypyl** [comment]
Does anybody know if this bug has been fixed

**Comment by rypyl** [comment]
is there a work around if we do not submit to app store - we are just distributing our application via enterprise certificate.

**Comment by basanthrzt** [comment]
Hi Guys,

Please let me know if there is a fix for this or Apple has already fixed it in its new update. Its **urgent**.

Thanks a ton!

**Comment by friedbunny** [comment]
No longer seeing this as of iOS 9.3.2.
